checker/src/main/java/org/checkerframework/checker/determinism/DeterminismQualifierPolymorphism.java

```diff
@@ -80,25 +83,65 @@ protected void replace(
                     qualHierarchy.leastUpperBounds(replacementsPolyDet, replacementsPolyDetNoOND);
             replacements = new AnnotationMirrorSet(lub);
         }
+        if (replacements == null) {
+            return;
+        }
 
         switch (value) {
             case "":
                 type.replaceAnnotations(replacements);
                 return;
             case "use":
-                if (!replacements.contains(factory.POLYDET)
-                        && !replacements.contains(factory.POLYDET_UP)
-                        && !replacements.contains(factory.POLYDET_DOWN)
-                        && !replacements.contains(factory.POLYDET_UPDET)) {
-                    type.replaceAnnotations(replacements);
+                if (replacementsPolyDet != null
+                        && replacementsPolyDet.contains(factory.POLYDET_NOORDERNONDET)) {
+                    type.replaceAnnotation(factory.POLYDET_USENOORDERNONDET);
+                }
+                // Replace @PolyDet("use") with @PolyDet if @PolyDet("use") doesn't
+                // resolve to a type that is a subtype of what @PolyDet resolves to.
```

Comment:
I thought `@PolyDet("use")` was suppose to resolve to the same annotation as `@PolyDet`.  Why is this only true if it is not a subtype?

Reply:
I had changed the semantics of `@PolyDet("use")` (after making collections invariant) as follows:
if the type that `@PolyDet(“use”)` resolves to is a subtype of the type that `@PolyDet `resolves to
then do nothing
else replace `@PolyDet(“use”)` with the type that `@PolyDet` resolves to.

Example:
`List.addAll()` is annotated as:
`@PolyDet("down") boolean addAll(@PolyDet List<@PolyDet("use") E> this, @PolyDet("use") Collection<? extends E> c);`

Consider the following example:
```
void test(
       @OrderNonDet List<@Det Integer> lst,
       @Det List<@Det Integer> lst1,
       @OrderNonDet List<@Det Integer> lst2) {
   lst.addAll(lst1);
   lst.addAll(lst2);
}
```

The above code is sound and type checks with this implementation of `@PolyDet(“use”)`
In method call `lst.addAll(lst1)`, `@PolyDet(“use”)` resolves to `@Det` which is a subtype of what `@PolyDet` resolves to (`@OrderNonDet`). Similarly for the method call `lst.addAll(lst2);`

If we add a call `lst.addAll(lst3)` where `lst3` is of the type `@NonDet List<@Det Integer>`, then `@PolyDet(“use”) `resolves to `@OrderNonDet` and the checker reports an error.

Reply:
Thanks, I have updated the manual. 
Note: the manual doesn't talk about invariant collections. I have left todos for future me and will update those sections when I create relevant PRs.